src/compas_fab/robots/robot.py

```diff
@@ -1014,6 +1068,114 @@ def inverse_kinematics(self, frame_WCF, start_configuration=None,
+            specified, including passive ones if available.
+        n_configurations : int, optional
+            Targeted number of desired configurations.
+        max_reach : float, optional
+            Maximum distance from the shoulder_link to the tool0_link of the selected group.
+            The max_reach value has to be lower than the maximum distance from the
+            shoulder_link to the tool0_link in order to force variation in the IK solver,
+            otherwise all returned configuations will be idential!
+            # TODO: for MoveIt IK solver, look at using seachPositionIK() instead of getPositionIK()?
+            #       Expose seachPositionIK() service?
+        shoulder_link_name : str, optional
+            Name of the robot's shoulder link, or other robot link name, that will be constrained
+            within the max_reach distance from the tool0_link.
+        timeout_seconds : float, optional
+            Timeout expressed in seconds
+
+        Raises
+        ------
+        compas_fab.backends.exceptions.BackendError
+            If no configuration can be found.
+
+        Returns
+        -------
+        list :class:`compas_fab.robots.Configuration`
+            The planning group's list of configurations within max_reach
+
+        Examples
+        --------
+        # TODO
+        """
+
+        if not group:
+            group = self.main_group_name
+        if shoulder_link_name is None:
+            raise NotImplementedError
+            # TODO: implement auto identification of the shoulder's link name for the current group, or main group
+
+        reach_constraints = self.position_constraint_from_max_reach(frame_WCF.point, shoulder_link_name, max_reach, group)
+        if constraints is None:
+            constraints = reach_constraints
+        else:
+            constraints.extend(reach_constraints)
+
+        result_configurations = []
+        wait_until = datetime.now() + timedelta(seconds=timeout_seconds)
+        break_loop = False
+        while not break_loop:
+            # NOTE: how can I catch "Error code: -31; NO_IK_SOLUTION" error without interrupting the loop?
+            try:
+                configuration = self.inverse_kinematics(frame_WCF,
+                                                        start_configuration=start_configuration,
+                                                        group=group,
+                                                        avoid_collisions=avoid_collisions,
+                                                        constraints=constraints,
+                                                        attempts=attempts,
+                                                        attached_collision_meshes=attached_collision_meshes,
+                                                        full_joint_state=full_joint_state)
+                result_configurations.append(configuration)
+            except BaseException:
```

Comment:
This should never be done, catching and passing `BaseException` will hide a lot of problems that should not be hidden.